packages/fx-core/src/component/resource/botService/appStudio/appStudioClient.ts

```diff
@@ -143,18 +148,24 @@ export class AppStudioClient {
   ): Promise<void> {
     const axiosInstance = AppStudioClient.newAxiosInstance(token);
 
-    let response = undefined;
     try {
-      response = await RetryHandler.Retry(() =>
+      const response = await RetryHandler.Retry(() =>
         axiosInstance.post(`${AppStudioClient.baseUrl}/api/botframework/${botReg.botId}`, botReg)
       );
+      if (!response || !response.data || response.status !== 200) {
```

Comment:
The following logic looks similar to that in createBotRegistration() method, is it possible to make it as a unified error handling, so that create/update/getBotRegistration can call it.

Reply:
Will do.